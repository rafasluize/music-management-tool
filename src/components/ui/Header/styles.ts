import styled from 'styled-components';

const Wrapper = styled.header`
  padding: ${({ theme }) => theme.spacing.km5} ${({ theme }) => theme.spacing.km8};
  display: flex;
  align-items: center;
  background-color: ${({ theme }) => theme.color.background.brand.default};
  background: linear-gradient(
    90deg,
    ${({ theme }) => theme.color.background.brand.default} 0%,
    ${({ theme }) => theme.color.background.secondary.default} 91%
  );

  h1 {
    color: ${({ theme }) => theme.color.text.neutral.inverted};
    font-size: ${({ theme }) => theme.font.size.m6};
  }
`;

const User = styled.div<{ letter: string }>`
  margin-left: auto;
  color: ${({ theme }) => theme.color.text.neutral.inverted};
  font-size: ${({ theme }) => theme.font.size.m4};

  &::after {
    content: '${({ letter }) => letter}';
    background-color: ${({ theme }) => theme.color.background.neutral.default};
    color: ${({ theme }) => theme.color.text.neutral.strong};
    font-weight: ${({ theme }) => theme.font.weight.bold};

    padding: ${({ theme }) => theme.spacing.km3};
    margin-left: ${({ theme }) => theme.spacing.km3};
    border-radius: ${({ theme }) => theme.border.radius.circle};
    width: 40px;
    height: 40px;
  }
`;

export { Wrapper, User };
