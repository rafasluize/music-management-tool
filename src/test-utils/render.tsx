import React, { FC, JSXElementConstructor, ReactElement, ReactNode, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { render, RenderOptions } from '@testing-library/react';
import theme from 'styles/theme';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter } from 'react-router-dom';

const ThemeDS: FC<{ children: ReactNode }> = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

export const wrapperRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) => render(component, { wrapper: ThemeDS, ...options });

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const queryRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  defaultValues?: unknown,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) =>
  render(
    <Suspense fallback={<p>Carregando</p>}>
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>{component}</BrowserRouter>
      </QueryClientProvider>
    </Suspense>,
    { wrapper: ThemeDS, ...options },
  );

export { wrapperRender as render, queryRender };

export default wrapperRender;
