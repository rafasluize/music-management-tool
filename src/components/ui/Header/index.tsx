import React, { FC } from 'react';

import * as Styled from './styles';
import { useNavigate } from 'react-router-dom';

const Header: FC = () => {
  const navigate = useNavigate();

  return (
    <Styled.Wrapper>
      <h1 onClick={() => navigate('/')}>Music Management Tools</h1>

      <Styled.User letter={'RL'}>Rafaela Luiz</Styled.User>
    </Styled.Wrapper>
  );
};

export default Header;
