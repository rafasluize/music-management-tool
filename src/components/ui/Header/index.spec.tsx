import React from 'react';
import { render } from 'test-utils/render';
import Header from '.';

describe('header', () => {
  it('snapshot', () => {
    const { container } = render(<Header />);

    // eslint-disable-next-line testing-library/no-node-access
    const wrapper = container.firstElementChild;

    expect(wrapper).toMatchSnapshot();
  });
});
