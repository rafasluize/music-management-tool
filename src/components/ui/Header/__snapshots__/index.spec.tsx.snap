// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`header snapshot 1`] = `
.c0 {
  padding: 20px 32px;
  display: flex;
  align-items: center;
  background-color: #dd2a62;
  background: linear-gradient(     90deg,     #dd2a62 0%,     #6c2c83 91%   );
}

.c0 h1 {
  color: #ffffff;
  font-size: 1.25rem;
}

.c1 {
  margin-left: auto;
  color: #ffffff;
  font-size: 1rem;
}

.c1::after {
  content: 'RL';
  background-color: #ffffff;
  color: #111111;
  font-weight: 700;
  padding: 12px;
  margin-left: 12px;
  border-radius: 100%;
  width: 40px;
  height: 40px;
}

<header
  class="c0"
>
  <h1>
    Music Management Tools
  </h1>
  <div
    class="c1"
    letter="RL"
  >
    Rafaela Luiz
  </div>
</header>
`;
